pkg/R/RDD.R

```diff
@@ -1294,6 +1294,79 @@ setMethod("sortBy",
             values(sortByKey(keyBy(rdd, func), ascending, numPartitions))
           })
 
+# Helper function to get first N elements from an RDD in the specified order.
+# Param:
+#   rdd An RDD.
+#   num Number of elements to return.
+#   ascending A flag to indicate whether the sorting is ascending or descending.
+# Return:
+#   A list of the first N elements from the RDD in the specified order.
+#
+takeOrderedElem <- function(rdd, num, ascending = TRUE) {          
+  if (num <= 0L) {
+    return(list())
+  }
+  
+  partitionFunc <- function(part) {
+    if (num < length(part)) {
+      # R limitation: order works only on primitive types!
+      ord <- order(unlist(part, recursive = FALSE), decreasing = !ascending)
+      part[ord[1:num]]
+    } else {
+      part
+    }
+  }
+  
+  newRdd <- mapPartitions(rdd, partitionFunc)
+  take(sortBy(newRdd, function(x) { x }, ascending = ascending), num)
```

Comment:
@shivaram , yes, I did look at the scala implementation. But I thought that code may have scalability issue. If you take a small number of elements from a large data set, that's OK. If you take a large number of elements from a large data set, the driver may suffer OOM (number of partitions * Num elements need to be transferred to the driver node) or it takes time to sort on the single driver node. Just a thought, no performance data proving:)
@concretevitamin , do you have any comments?
I am fine to follow the scala implementation if you prefer that.